### PR TITLE
fix(skywalking-logger, error-log-logger): skywalking service_instance_name $hostname is supported

### LINKED ISSUE #7251: request help: skywalking-logger and error-log-logger plugin service_instance_name set $hostname invalid
### Current Behavior

skywalking-logger and error-log-logger plugins set service_instance_name to `$hostname` and report to skywalking

instance is represented as `"$hostname"`string

image

### Expected Behavior

Expect service instances to be a string of $hostname instead of "$hostname" after reporting to Skywalking

image

### Error Logs

no error logs

### Steps to Reproduce

We can refer to the skywalking plugin processing method

https://github.com/apache/apisix/blob/45816272a80d6d2045259d32acdabbec4039ed1e/apisix/plugins/skywalking.lua#L130-L132

### Environment

- APISIX version (run `apisix version`): 2.13.1
- Operating system (run `uname -a`): Linux apisix-844457679-8242t 5.10.102.1-microsoft-standard-WSL2+ #1 SMP Mon Jun 13 04:56:19 UTC 2022 x86_64 Linux
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): openresty/1.19.9.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`): 
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by dickens7**
### Description

Fixes #7251

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/error-log-logger.lua [hunk 1]
```diff
@@ -180,10 +180,15 @@ local function send_to_skywalking(log_message)
     httpc:set_timeout(config.timeout * 1000)
 
     local entries = {}
+    local serviceInstanceName = config.skywalking.service_instance_name
```
Comment (spacewander) [inline apisix/plugins/error-log-logger.lua:183]: We should call it in the `init` method, instead of in the per-report way

**Review by spacewander — COMMENTED** [review]
Let's submit the PR to master, and backport it after it's merged

File: apisix/plugins/error-log-logger.lua (hunk 1 above)
Comment (spacewander) [reply apisix/plugins/error-log-logger.lua:183]: I change my mind. It is a route-level configuration and may be changed on the fly.

**Comment by tokers** [comment]
Can we add some test cases to cover the changes?

**Comment by dickens7** [comment]
> Can we add some test cases to cover the changes?

I don't know how this test case can be written to cover this scenario.

Shall we query the data received by skywaling?

**Comment by soulbird** [comment]
> > Can we add some test cases to cover the changes?
> 
> I don't know how this test case can be written to cover this scenario.
> 
> Shall we query the data received by skywaling?

Just for reference:  https://github.com/apache/apisix/blob/master/t/plugin/skywalking-logger.t#L124-L196.
However, in the instance of executing the github action, the hostname may not be determined

**Comment by tzssangglass** [comment]
we can verify `hostname` like: https://github.com/apache/apisix/blob/master/t/plugin/server-info.t#L108

**Comment by dickens7** [comment]
> we can verify `hostname` like:
> 
> https://github.com/apache/apisix/blob/afce02cea506eb4c70f2dd9d5230d18c7ab8b51e/t/plugin/server-info.t#L108

This won't work, because `$hostname` can also be matched but we expect the value to be something other than `$hostname`

**Comment by tzssangglass** [comment]
how about `^[a-zA-Z\-0-9\.]+`

**Comment by dickens7** [comment]
> how about `^[a-zA-Z\-0-9\.]+`

`service_instance_name` is user-defined and can be set to any value such as `#xxx`.
This is just for testing, it should work.

But I think my current use case should be ok, we just need to judge that service_instance_name is not equal to the string `$hostname` and not equal to the empty string.

```
--- no_error_log eval
qr/\\\"serviceInstance\\\":\\\"\$hostname\\\"/
qr/\\\"serviceInstance\\\":\\\"\\\"/
```

**Review by spacewander — APPROVED** [review]

**Comment by tzssangglass** [comment]
I reruned the failed CI and if CI passed, we can merge it.

**Review by tzssangglass — APPROVED** [review]

File: apisix/plugins/error-log-logger.lua (hunk 1 above)
Comment (starsz) [inline apisix/plugins/error-log-logger.lua:183]: IMO, we should use this code style, like "service_instance_name`

Comment (dickens7) [reply apisix/plugins/error-log-logger.lua]: Yes, we should use snake case naming

**Comment by dickens7** [comment]
please rerun ci

**Review by spacewander — APPROVED** [review]

**Review by starsz — APPROVED** [review]

File: apisix/plugins/error-log-logger.lua [hunk 2]
```diff
@@ -180,10 +180,15 @@ local function send_to_skywalking(log_message)
     httpc:set_timeout(config.timeout * 1000)
 
     local entries = {}
+    local service_instance_name = config.skywalking.service_instance_name
+    if service_instance_name == "$hostname" then
+        service_instance_name = core.utils.gethostname()
+    end
```
Comment (leslie-tsang) [inline apisix/plugins/error-log-logger.lua:186]: Hello there, we need a test case to ensure it work as expected. :)

**Review by leslie-tsang — COMMENTED** [review]
LGTM. 👍

**Pull request merged**